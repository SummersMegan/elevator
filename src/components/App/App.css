*{
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  border: 0;
}

nav{
  background-color: #2d2825;
  color: #517d59;
  height: 60px;
  font-size: 28px;
  text-align: center;
  line-height: 60px;
}

main{
  margin: 20px;
  border-bottom: 5px solid black;
  height: 460px;
  display: flex;
  justify-content: center;
}

footer{
  font-size: 12px;
  text-align: center;
  padding-bottom: 10px;
}

footer button{
  padding: 10px;
  margin-bottom: 5px;
  background-color: #517d59;
  color: white;
  cursor: pointer;
}