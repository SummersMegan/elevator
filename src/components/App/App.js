import React, { Component } from 'react'
import './App.css'
import FloorSelector from '../FloorSelector/FloorSelector'
import FloorTracker from '../FloorTracker/FloorTracker'
import ElevatorMusic from '../../assets/elevator-music.mp3'

class App extends Component{
  
  constructor(props){
    super(props)
    this.state={
      allFloors:[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30],
      usersSelectedFloor:null,
      selectedFloors:[],
      currentFloor:1,
      play: false
    }
    this.audio=new Audio(ElevatorMusic)
  }

  playMusic=()=>{
    this.setState({ 
        play: true 
      })
      this.audio.play()
    }
    
  pauseMusic=()=>{
    this.setState({ 
      play: false 
    })
    this.audio.pause()
  }

  changeCurrentFloor=(newFloor)=>{
    this.setState({
      currentFloor: newFloor
    })
  }

  handleUserSelectsFloor=(floor)=>{
    //if user did not already select floor    
    if(floor!==this.state.usersSelectedFloor){
      this.setState(prevState=>({
        usersSelectedFloor:floor,
        selectedFloors:[...prevState.selectedFloors,floor]
      }))
    }
  }

  renderMusicButtons=()=>{
    if(this.state.play===true){
      return(
        <button className='musicButton' onClick={this.pauseMusic}>Pause Music</button>
      )
    } else{
      return(
        <button className='musicButton' onClick={this.playMusic}>Play Elevator Music</button>
      )
    }
  }

  render(){
    return(
      <>
      <nav>
        <h1>Elevator Ride</h1>
      </nav>
      <main>
        <FloorTracker state={this.state} handleFloorChange={this.changeCurrentFloor}/>
        <FloorSelector state={this.state} handleUserSelectsFloor={this.handleUserSelectsFloor}/>
      </main>
      <footer>
        {this.renderMusicButtons()}
        <p>Royalty Free Music from Bensound</p>
      </footer>
      </>
    )
  }
  
}

export default App
