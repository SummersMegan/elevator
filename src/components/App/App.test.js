import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><App/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <App/>
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
})