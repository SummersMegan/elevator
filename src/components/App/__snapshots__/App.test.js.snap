// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
Array [
  <nav>
    <h1>
      Elevator Ride
    </h1>
  </nav>,
  <main>
    <div
      className="floorTrackerContainer"
    >
      <div
        className="elevator"
      >
        <p
          className="floorTracker"
        >
          1
        </p>
        <div
          className="elevatorDoors"
        >
          <div />
          <div />
        </div>
      </div>
    </div>
    <div
      className="floorSelectorContainer"
    >
      <h3>
        Please select a floor!
      </h3>
      <div
        className="buttonContainer"
      >
        <button
          onClick={[Function]}
        >
          1
        </button>
        <button
          onClick={[Function]}
        >
          2
        </button>
        <button
          onClick={[Function]}
        >
          3
        </button>
        <button
          onClick={[Function]}
        >
          4
        </button>
        <button
          onClick={[Function]}
        >
          5
        </button>
        <button
          onClick={[Function]}
        >
          6
        </button>
        <button
          onClick={[Function]}
        >
          7
        </button>
        <button
          onClick={[Function]}
        >
          8
        </button>
        <button
          onClick={[Function]}
        >
          9
        </button>
        <button
          onClick={[Function]}
        >
          10
        </button>
        <button
          onClick={[Function]}
        >
          11
        </button>
        <button
          onClick={[Function]}
        >
          12
        </button>
        <button
          onClick={[Function]}
        >
          13
        </button>
        <button
          onClick={[Function]}
        >
          14
        </button>
        <button
          onClick={[Function]}
        >
          15
        </button>
        <button
          onClick={[Function]}
        >
          16
        </button>
        <button
          onClick={[Function]}
        >
          17
        </button>
        <button
          onClick={[Function]}
        >
          18
        </button>
        <button
          onClick={[Function]}
        >
          19
        </button>
        <button
          onClick={[Function]}
        >
          20
        </button>
        <button
          onClick={[Function]}
        >
          21
        </button>
        <button
          onClick={[Function]}
        >
          22
        </button>
        <button
          onClick={[Function]}
        >
          23
        </button>
        <button
          onClick={[Function]}
        >
          24
        </button>
        <button
          onClick={[Function]}
        >
          25
        </button>
        <button
          onClick={[Function]}
        >
          26
        </button>
        <button
          onClick={[Function]}
        >
          27
        </button>
        <button
          onClick={[Function]}
        >
          28
        </button>
        <button
          onClick={[Function]}
        >
          29
        </button>
        <button
          onClick={[Function]}
        >
          30
        </button>
      </div>
    </div>
  </main>,
  <footer>
    <button
      className="musicButton"
      onClick={[Function]}
    >
      Play Elevator Music
    </button>
    <p>
      Royalty Free Music from Bensound
    </p>
  </footer>,
]
`;
